src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -911,7 +936,6 @@ ConvertNonExistingPlacementDDLCommandsToTasks(List *shardCommandList,
 		SetPlacementNodeMetadata(taskPlacement, workerNode);
 
 		task->taskPlacementList = list_make1(taskPlacement);
-		task->anchorShardId = shardId;
```

Comment:
This was an important change to make shard splits work, because the shard split shards don't exist in the metadata yet.